app/api/resources/grpc_client.py

```diff
@@ -85,4 +85,6 @@ def get_region(self, genome_uuid: str, region_name: str):
     def get_genome_uuid_from_slug(self, slug):
         uuid_request = ensembl_metadata_pb2.GenomeTagRequest(genome_tag=slug)
         genome_uuid_data = self.stub.GetGenomeUUIDByTag(uuid_request)
-        return genome_uuid_data.genome_uuid
\ No newline at end of file
+        if genome_uuid_data.genome_uuid:
+            return genome_uuid_data.genome_uuid
+        return slug
```

Comment:
Also, the `get_genome_uuid_from_slug` is probably the wrong function to return the `slug` from. In pseudocode, I would imagine the following conversation with your code:

- Is this string I received from the client a genome_tag?
  - Yes, it it; cool; I know genome uuid now.
  - No, it isn't
- Hmm, I now either have a certain genome uuid resolved by genome_tag; or the original string from client that can be a genome uuid, or can be rubbish. Let's try and fetch genome info using the genome uuid that I've got.
  - Did I receive genome info from grpc? Great, return response according to spec
  - Did I not receive genome info? Then send back a 404 response.

Reply:
Yes, Just refresh the page :) I just implemented and pushed the changes :)

Reply:
It's just that I would expect the function called `get_genome_uuid_from_slug` to work differently — to return a real genome uuid the string is a genome tag, or None if it isn't (and also, probably to be called `get_genome_uuid_from_genome_tag` for clarity). As it is written, it returns either a real genome uuid or a garbage string.

But it is up to you to choose which is better.